node/node_handler.go

```diff
@@ -506,26 +477,37 @@ func (node *Node) PostConsensusProcessing(
 	}
 }
 
-// bootstrapConsensus is the a goroutine to check number of peers and start the consensus
-func (node *Node) bootstrapConsensus() {
-	tick := time.NewTicker(5 * time.Second)
-	defer tick.Stop()
-	for range tick.C {
-		numPeersNow := node.host.GetPeerCount()
-		if numPeersNow >= node.Consensus.MinPeers {
-			utils.Logger().Info().Msg("[bootstrap] StartConsensus")
-			node.startConsensus <- struct{}{}
-			return
+// BootstrapConsensus is the a goroutine to check number of peers and start the consensus
+func (node *Node) BootstrapConsensus() error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	min := node.Consensus.MinPeers
+	needFriends := make(chan struct{})
```

Comment:
Name `needFriends` is confusing. Maybe we can find a better name for this?

Reply:
renamed to enoughMinPeers